lib/ansible/modules/clustering/k8s/k8s_event.py

```diff
@@ -0,0 +1,342 @@
+     message:
+       description: Status for operation
+       returned: success
+       type: str
+     metadata:
+       description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+       returned: success
+       type: complex
+     reason:
+       description: Reason for the transition into the objects current status
+       returned: success
+       type: str
+     reportingComponent:
+       description: Component responsible for event
+       returned: success
+       type: str
+     source:
+       description: Source contains information for an Event
+       returned: success
+       type: complex
+     type:
+       description: Type of Event. Either Normal or Warning
+       returned: success
+       type: string
+
+"""
+
+import copy
+import traceback
+import datetime
+import kubernetes.config.dateutil
+import openshift
+
+from collections import defaultdict
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+EVENT_ARG_SPEC = {
+    "state": {"default": "present", "choices": ["present", "absent"]},
+    "name": {"required": True},
+    "namespace": {"required": True},
+    "merge_type": {"type": "list", "choices": ["json", "merge", "strategic-merge"]},
+    "message": {"type": "str", "required": True},
+    "reason": {"type": "str", "required": True},
+    "reportingComponent": {"type": "str", "required": True},
+    "type": {"choices": ["Normal", "Warning"]},
+    "source": {
+        "type": "dict",
+        "component": {"type": "str", "required": True}
+    },
+    "involvedObject": {
+        "type": "dict",
+        "apiVersion": {"type": "str", "required": True},
+        "kind": {"type": "str", "required": True},
+        "name": {"type": "str", "required": True},
+        "namespace": {"type": "str", "required": True},
+    },
+}
+
+
+class KubernetesEvent(KubernetesRawModule):
+    def __init__(self, *args, **kwargs):
+        super(KubernetesEvent, self).__init__(*args, k8s_kind="Event", **kwargs)
+
+    @property
+    def argspec(self):
+        """ argspec property builder """
+        argumentSpec = copy.deepcopy(AUTH_ARG_SPEC)
+        argumentSpec.update(EVENT_ARG_SPEC)
+        return argumentSpec
+
+    def execute_module(self):
+        """ Module execution """
+        self.client = self.get_api_client()
+
+        message = self.params.get("message")
+        reason = self.params.get("reason")
+        reporting_component = self.params.get("reportingComponent")
+        event_type = self.params.get("type")
+        source = self.params.get("source")
+
+        definition = defaultdict(defaultdict)
+
+        meta = definition["metadata"]
+        meta["name"] = self.params.get("name")
+        meta["namespace"] = self.params.get("namespace")
+
+        involved_obj = self.params.get("involvedObject")
+
+        resource = self.find_resource("Event", "v1", fail=True)
+
+        prior_event = None
```

Comment:
This line of code is not required, you can modify your condition @ line 293 as "prior_event.get('reason') == reason"